client/web/compose/src/components/PageBlocks/Configurator.vue

```diff
@@ -445,6 +477,24 @@ export default {
         this.$set(this.block.meta.visibility, 'roles', roles)
       },
     },
+
+    autoCompleteSuggestionTree () {
+      const moduleFields = (this.module.fields || []).map(({ name }) => name)
+      const userProperties = this.$auth.user.properties()
+      const recordProperties = this.record ? this.record.properties() : undefined
+      const recordValueFields = this.record ? Object.keys(this.record.values) : undefined
+
+      return getRecordBasedSuggestions({ moduleFields, userProperties, recordProperties, isRecordPage: this.isRecordPage, recordValueFields })
+    },
+
+    visibilityAutoSuggestionTree () {
+      const moduleFields = (this.module.fields || []).map(({ name }) => name)
```

Comment:
Same here, maybe makes sense to add this to a lib, since the same thing will be there a bunch of times.

Maybe make it part of the suggestion tree since there is 3 types of expressions I think

1. Record filters
2. Field expressions/validators
3. Visibility expressions
4. Layout expressions